storage/src/mmr/benches/update.rs

```diff
@@ -11,41 +11,57 @@ fn bench_update(c: &mut Criterion) {
+                        module_path!(),
+                        updates,
+                        leaves,
+                        batched
+                    ),
+                    |b| {
+                        b.to_async(&runner).iter_custom(|_iters| async move {
+                            let mut mmr = Mmr::<Sha256>::new();
+                            let mut elements = Vec::with_capacity(leaves as usize);
+                            let mut sampler = StdRng::seed_from_u64(0);
+                            let mut leaf_positions = Vec::with_capacity(leaves as usize);
+                            let mut h = Sha256::new();
+                            let mut h = Standard::new(&mut h);
 
-                        // Append random elements to MMR
-                        for _ in 0..leaves {
-                            let digest = sha256::Digest::random(&mut sampler);
-                            elements.push(digest);
-                            let pos = mmr.add(&mut h, &digest).await.unwrap();
-                            leaf_positions.push(pos);
-                        }
+                            // Append random elements to MMR
+                            for _ in 0..leaves {
+                                let digest = sha256::Digest::random(&mut sampler);
+                                elements.push(digest);
+                                let pos = mmr.add(&mut h, &digest).await.unwrap();
+                                leaf_positions.push(pos);
+                            }
 
-                        // Randomly update leaves -- this is what we are benchmarking.
-                        let start = Instant::now();
-                        for _ in 0..updates {
-                            let rand_leaf_num = sampler.gen_range(0..leaves);
-                            let rand_leaf_pos = leaf_positions[rand_leaf_num as usize];
-                            let rand_leaf_swap = sampler.gen_range(0..elements.len());
-                            let new_element = &elements[rand_leaf_swap];
-                            mmr.update_leaf(&mut h, rand_leaf_pos, new_element)
-                                .await
-                                .unwrap();
-                        }
-
-                        start.elapsed()
-                    });
-                },
-            );
+                            // Randomly update leaves -- this is what we are benchmarking.
+                            let start = Instant::now();
+                            for _ in 0..updates {
+                                let rand_leaf_num = sampler.gen_range(0..leaves);
+                                let rand_leaf_pos = leaf_positions[rand_leaf_num as usize];
+                                let rand_leaf_swap = sampler.gen_range(0..elements.len());
+                                let new_element = &elements[rand_leaf_swap];
+                                if batched {
```

Comment:
nit: in this case I like `if batched else` instead of `if batched continue`

Reply:
changed